update NucleationMicroForce with model 2022 (#149)
hugary1995/raccoon: doc/content/source/materials/NucleationMicroForce.md

+\frac{\sqrt{3}\sigma_{ts}}{6(3\lambda+2\mu)}+\frac{\sqrt{3}\sigma_{ts}}{6\mu}.
\end{equation}

The $I_1$ and $J_2$ are invariants for the degraded Cauchy stress:

\begin{equation}
    I_1=tr(\boldsymbol{\sigma}), \quad
\end{equation}

\begin{equation}
    J_2=\frac{1}{2}(\boldsymbol{\sigma}-\frac{1}{3}(tr(\boldsymbol{\sigma})\boldsymbol{I})):(\boldsymbol{\sigma}-\frac{1}{3}(tr(\boldsymbol{\sigma})\boldsymbol{I})).
\end{equation}

The numerical stress envelope introduced by the external driving force is

\begin{equation}
\hat{F}^{l_0}(I_1,J_2)=\frac{J_2}{\mu}+\frac{I_1^2}{9\kappa}-c_e(I_1,J_2;l_0)-\frac{3G_c}{8l_0}=0,
\end{equation}

which strictly satisfied the critical tensile strength $\sigma_{ts}$ and critical compressive strength $\sigma_{cs}$. Other locations on the surface approach to an existed model for the stress envelope, the Drucker-Prager model[!cite](https://www.jstor.org/stable/43633942), as regularization length approach zero limit:

\begin{equation}
    \hat{F}(I_1,J_2)
    =\sqrt{J_2}
    +\frac{\sigma_{cs}-\sigma_{ts}}{\sqrt{3}(\sigma_{cs}+\sigma_{ts})}I_1
    -\frac{2\sigma_{cs}\sigma_{ts}}{\sqrt{3}(\sigma_{cs}+\sigma_{ts})}I_1=0
\end{equation}

> $l_0$ is the prescribed regularization length. By enforcing the irreversibility restriction only when the material is close to fully damaged, say $d\ge 0.95$, the $G_c$ behavior of Griffith model can be kept by calibrating parameter $\delta$. Currently we use a [surfing boundary problem](tutorials/12_surfing_boundary_problem.md) to match the precribed $G_c$ and the J intergral. Be aware that this $\delta$ is currently dependent on material properties and model parameters (regularization length, mesh size,...) until further developed version. Do calibrate it when any of the parameter is changed.

Reviewer: ```suggestion
$l_0$ is the prescribed regularization length. By enforcing the irreversibility restriction only when the material is close to fully damaged, say $d\ge 0.95$, the $G_c$ behavior of Griffith model can be kept by calibrating parameter $\delta$. Currently we use a [surfing boundary problem](tutorials/12_surfing_boundary_problem.md) to match the precribed $G_c$ and the J intergral. Be aware that this $\delta$ is currently dependent on material properties and model parameters (regularization length, mesh size,...). In other words, calibration of $\delta$ is necessary.
```

During crack propagation, the effective regularization length $l_e$ 

\begin{equation}
l_e = \frac{l_0}{\sqrt{1+\delta}},\quad (\delta > -1)
\end{equation}

> could be different than the prescribed one. Make sure your mesh size can also fully resolve it.

Reviewer: ```suggestion
could deviate from $\ell$. The mesh should be able to resolve $l_e$ as well.
```
